docs/docs/configuration/pages.md

```diff
@@ -95,7 +97,19 @@ export default function SignIn({ providers }) {
 }
 
 export async function getServerSideProps(context) {
-  const providers = await getProviders()
+  const session = await getServerSession(context.req, context.res, authOptions);
+  
+  // If the user is already logged in, redirect,
+  // unless the destination is on the same page,
+  // to prevent infinite redirects
+  const destination = "/"
+  // Note, this does not check hash (#) or search params (?)
+  if (session && destination !== authOptions.pages?.signIn) {
+    return { redirect: { destination } };
+  }
```

Comment:
```suggestion
  // If the user is already logged in, redirect.
  // Note: Make sure not to redirect to the same page
  // To avoid an infinite loop!
  if (session) {
    return { redirect: { destination: "/" } };
  }
```